app.js

```diff
@@ -102,4 +102,93 @@ app.delete('/api/v1/palettes/:id', async (request, response) => {
   }
 });
 
+app.get('/api/v1/palettes/:id', async (request, response ) => {
+  const { id } = request.params;
+  try {
+    const palette = await database('palettes').where('id', id);
+
+    if (!palette.length) {
+      return response.status(404).json({ error: `Could not find a palette with id: ${id}`})
+    }
+
+    response.status(200).json({
+      id: palette[0].id,
+      title: palette[0].title,
+      color_1_id: palette[0].color_1_id,
+      color_2_id: palette[0].color_2_id,
+      color_3_id: palette[0].color_3_id,
+      color_4_id: palette[0].color_4_id,
+      color_5_id: palette[0].color_5_id,
+      project_id: palette[0].project_id
+    });
+  } catch (error) {
+    response.status(500).json({ error })
+  }
+});
+
+app.get('/api/v1/palettes', async (request, response) => {
+  try {
+    const palettes = await database('palettes').select();
+    const displayPalettes = palettes.map(palette => ({
+      id: palette.id,
+      title: palette.title,
+      color_1_id: palette.color_1_id,
+      color_2_id: palette.color_2_id,
+      color_3_id: palette.color_3_id,
+      color_4_id: palette.color_4_id,
+      color_5_id: palette.color_5_id,
+      project_id: palette.project_id
+    }));
+
+    response.status(200).json({ palettes: displayPalettes });
+  } catch (error) {
+    response.status(500).json({ error });
+  }
+});
+
+app.post('/api/v1/palettes', async (request, response) => {
+  const palette = request.body;
+
+  for (let requiredParameter of ['title', 'color_1_id', 'color_2_id', 'color_3_id', 'color_4_id', 'color_5_id', 'project_id']) {
+    if (!palette[requiredParameter]) {
+      return response.status(422).send({ error: `Expected format: { title: <String>, color_1_id: <String>, color_1_id: <String>, color_1_id: <String>, color_1_id: <String>, color_1_id: <String>, project_id: <Integer> }. You're missing a "${requiredParameter}" property.` });
+    }
```

Comment:
Now that I'm looking at the color properties, I'm realizing that I failed to adjust the key names to to remove '_id' after we switched from id's to hex colors. My bad.

I'll make a note to consider reformatting the key names as a late refactor/ chore, but it's not that important

Reply:
Are you suggesting we make the response object keys say `color_1_hex` rather than `color_1_id`?